client/web/web.go

```diff
@@ -218,23 +214,23 @@ func NewServer(opts ServerOpts) (s *Server, err error) {
 	return s, nil
 }
 
-func (s *Server) withCSRF(h http.Handler) http.Handler {
-	csrfProtect := csrf.Protect(s.csrfKey(), csrf.Secure(false))
-
-	// ref https://github.com/tailscale/tailscale/pull/14822
-	// signal to the CSRF middleware that the request is being served over
-	// plaintext HTTP to skip TLS-only header checks.
-	withSetPlaintext := func(h http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			r = csrf.PlaintextHTTPRequest(r)
-			h.ServeHTTP(w, r)
-		})
-	}
+// withSecFetchSite wraps a HTTP handler and ensures requests made to it have
+// the `Sec-Fetch-Site` header set to same-origin to prevent CSRF attacks.
+func (s *Server) withSecFetchSite(h http.Handler) http.Handler {
+	// Ref https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Sec-Fetch-Site
+	// Sent by all modern browsers to indicate the relationship between the
+	// origin from which the request was made and the requested resource.
+	// Require that the Sec-Fetch-Site header is set to "same-origin"
+	// for all requests to the web client, preventing CSRF attacks.
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Sec-Fetch-Site") != "same-origin" {
+			http.Error(w, "forbidden", http.StatusForbidden)
```

Comment:
let's return a different error when `Sec-Fetch-Site` is not set.
also, perhaps make this error slightly more verbose than `"forbidden"`